src/Symfony/Component/HttpKernel/Event/RequestEvent.php

```diff
@@ -20,6 +22,37 @@
  *
  * @author Bernhard Schussek <bschussek@gmail.com>
  */
-class RequestEvent extends GetResponseEvent
+class RequestEvent extends KernelEvent
 {
+    private $response;
+
+    /**
+     * Returns the response object.
+     *
+     * @return Response|null
+     */
+    public function getResponse()
+    {
+        return $this->response;
+    }
+
+    /**
+     * Sets a response and stops event propagation.
+     */
+    public function setResponse(Response $response)
+    {
+        $this->response = $response;
+
+        $this->stopPropagation();
+    }
+
+    /**
+     * Returns whether a response was set.
+     *
+     * @return bool Whether a response was set
```

Comment:
let's use a real type instead? could require making either the class or the method final in 4.4